tools/invocation_header_gen.py

```diff
@@ -184,6 +177,15 @@ def generate(args, invocations):
 
 
 if __name__ == "__main__":
+    # We require jinja2 to be at least version 2.10,
+    # In the past we used the 'namespace' feature from that version.
+    # other versions of jinja, particularly `minijinja`, don't support
+    # namespaces. However in case `namespace` is needed in the future require a
+    # version which supports it.
+    jinja2_version = version("jinja2")
+    if jinja2_version < "2.10":
```

Comment:
Can't we do this check at the place where we did it before, so this will always fail regardless how this file is used? Otherwise, instead of putting this here, it might be better to put it at the place where we need this specific feature then.

Reply:
Done. Move it to the function that actually use jinja2 methods